components/ContactForm/index.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState, useRef, useEffect } from "react";
+import { RenderMsg } from "./renderMsg";
+import { isEmailValid, isMessageValid, isNameValid } from "./validations";
+import styles from "./Styles.module.css";
+
+type FormState = {
+  name: string;
+  email: string;
+  message: string;
+};
+
+export const ContactForm = () => {
+  const [nameErr, setNameErr] = useState<string | null>(null);
+  const [emailErr, setEmailErr] = useState<string | null>(null);
+  const [messageErr, setMessageErr] = useState<string | null>(null);
+  const [formState, setstate] = useState<FormState>({
+    name: "",
+    email: "",
+    message: "",
+  });
+
+  const nameInput = useRef<HTMLInputElement>(null);
+  const emailInput = useRef<HTMLInputElement>(null);
+  const messageInput = useRef<HTMLTextAreaElement>(null);
+
+  useEffect(() => {
+    if (nameInput.current != null) {
+      nameInput.current.focus();
+    }
+  }, []);
+
+  useEffect(() => {
+    const inputToFocus = document.querySelector(
+      "." + styles.error
+    ) as HTMLElement;
+    if (inputToFocus !== null) {
+      inputToFocus.focus();
+    }
+  }, [nameErr, emailErr, messageErr]);
+
+  const handleChange = (
+    event:
+      | React.ChangeEvent<HTMLInputElement>
+      | React.ChangeEvent<HTMLTextAreaElement>
+  ) => {
+    setstate({ ...formState, [event.target.name]: event.target.value });
+  };
+
+  const formIsValid = () => {
+    const nameValid = isNameValid(formState.name);
+    const messageValid = isMessageValid(formState.message);
+    const emailValid = isEmailValid(formState.email);
+    setNameErr(nameValid);
+    setMessageErr(messageValid);
+    setEmailErr(emailValid);
+
+    const errArray = [nameValid, messageValid, emailValid];
+    let isValid = true;
```

Comment:
A better approach from functional programming perspective would be to use `errArray.some` or `errArray.every`